### DISCUSSION TITLE
How the degree of de trending is determined in PyCaret for time series?

### DISCUSSION THREAD

**Discussion by debdarsan** [pycaret/pycaret#2443] [Q&A]
Hi,

I am using pycaret for time series [_**pycaret-ts-alpha**_ (version 3.0.0.dev1648160839)]. I have a data set that is non-stationary, which can be converted to a non-stationary time series by taking the first-order difference. I am examining two scenarios:

**_A_.** Feed the non-stationary raw data to the 'setup' function
*_*B_.** Convert the non-stationary raw data to stationary time series by taking the first-order difference and feeding that to the 'setup' function

Now, I am looking at the results the "**check_stats()**" function returned which is described below:

For case A: The values obtained were '**FALSE**' for the properties 'Stationarity', 'Trend Stationarity' and 'Normality'
For case B: The values obtained were '**TRUE**' for the properties 'Stationarity', 'Trend Stationarity' and 'Normality'

We can see the above results returned by PyCaret are correct and what were to be expected. No problem at all.

However, when I am examining the model _PyCaret_ is using for training, wherever the '_BaseCDSDtForecaster_' wrapper (conditional de-seasonalizing and de-trending) is used, a 'degree' value of 1 is used for both the cases A and B. I would have expected that degree to have a value of 1 when the input data is non-stationary (case B) and a value of zero when the data is already de-trended (case A).

I can understand that during tuning, the best value is obtained by doing a random search and the parameter 'degree' can assume any suitable value (1, 2, 3 etc.), but I have a gap in my understanding of why in the 'training' phase the value is 1 always, no matter if the data has a trend or not.

Please help.

Dr Debdarsan Niyogi

**Comment by ngupta23** [comment]
@debdarsan Good observation. I am glad someone is examining these in detail :)

The idea with CdsDt models was that even if the model is stationary, performing detrending (with default order 1) would likely not harm as the result would still be stationary. One could argue that in this case, the order should be 0. My problem with using ADF test to make any automated decisions like this is that ADF test is a low-power test and can give false results more often than expected. 

Also, another argument for this would be that for example AR(1) with a high value of `phi` (say 0.95) can be modeled as both stationary and non-stationary models (in an automated system) without any further information about the process generating the data. This would likely produce similar fit statistics. Predictions will however be different (with one reverting to mean and the other just predicting the last point forward, but an automated system would have no way of knowing this without human interference). 

But your comments are fair. Maybe this model's functionality needs to be improved further taking into account the stationarity (even if it means using the ADF test). I will move this discussion into an Issue and we can continue hashing it out there.

**Comment by debdarsan** [comment]
@ngupta23 Any update on this?
